src/components/BaseSeparatePage/TableOfContents/index.tsx

```diff
@@ -1,50 +1,115 @@
-import React, { FC } from 'react'
-import clsx from 'clsx'
-import { StyledLink } from '../../StyledLink'
-import TableOfContent from './TableOfContent'
+import React, {
+  ComponentProps,
+  createContext,
+  FC,
+  PropsWithChildren,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
 
-import GithubIcon from './githubIcon.svg'
-import DownArrow from './downArrow.svg'
+export interface TOCItemInfo {
+  id: string
+  title: string
+  el: HTMLElement
+  isActive: boolean
+  isInViewport: boolean
+}
+
+export interface TOCContextValue {
+  tocItems: TOCItemInfo[]
+  addTOCItem: (tocItemEl: HTMLElement, title: string) => void
+  removeTOCItem: (tocItemEl: HTMLElement) => void
+}
+
+export const TOCContext = createContext<TOCContextValue>(
```

Comment:
TOCContext and TOCItem should be elevated to a public component that provides the base TOC logic and then internally wraps a new StyledTOCList based on it wherever needed (e.g. in BaseSeparatePage) .